habitat-lab/habitat/datasets/rearrange/navmesh_utils.py

```diff
@@ -0,0 +1,441 @@
+        self.vel_control.linear_velocity = mn.Vector3(
+            [0.0, 0.0, -linear_velocity]
+        )
+        self.vel_control.angular_velocity = mn.Vector3(
+            [0.0, angular_velocity, 0.0]
+        )
+        # Compute the rigid state
+        rigid_state = habitat_sim.RigidState(
+            mn.Quaternion.from_matrix(trans.rotation()), trans.translation
+        )
+        # Get the target rigid state based on the simulation frequency
+        target_rigid_state = self.vel_control.integrate_transform(
+            1 / self._sim_freq, rigid_state
+        )
+        # Get the ending pos of the agent
+        end_pos = target_rigid_state.translation
+        # Offset the height
+        end_pos[1] = trans.translation[1]
+        # Construct the target trans
+        target_trans = mn.Matrix4.from_(
+            target_rigid_state.rotation.to_matrix(),
+            target_rigid_state.translation,
+        )
+
+        return target_trans
+
+
+def is_navigable_given_robot_navmesh(
+    sim,
+    start_pos,
+    goal_pos,
+    navmesh_offset=([0, 0], [0, 0.15], [0, -0.15]),
+    angle_threshold=0.05,
+    angular_velocity=1.0,
+    distance_threshold=0.25,
+    linear_velocity=1.0,
+    vdb=None,
+    render_debug_video=False,
+):
+    """
+    Return True if the robot can navigate from point A
+    to point B given the configuration of the navmesh.
+    """
+    logger.info(
+        "Checking robot navigability between target object start and goal:"
+    )
+
+    # TODO: for now computation is repeated here, but should be provided from earlier
+    largest_island_id = get_largest_island_index(
+        sim.pathfinder, sim, allow_outdoor=False
+    )
+
+    snapped_start_pos = sim.pathfinder.snap_point(start_pos, largest_island_id)
+    snapped_goal_pos = sim.pathfinder.snap_point(goal_pos, largest_island_id)
+
+    logger.info(
+        f"     - start_pos to snapped_start_pos distance = {(snapped_start_pos-start_pos).length()}"
+    )
+    logger.info(
+        f"     - goal_pos to snapped_goal_pos distance = {(snapped_goal_pos-goal_pos).length()}"
+    )
+
+    if render_debug_video:
+        assert vdb is not None, "Need a vdb for visual debugging."
+        sim.navmesh_visualization = True
+
+    # Create a new pathfinder with slightly stricter radius to provide nav buffer from collision
+    pf = habitat_sim.nav.PathFinder()
+    modified_settings = sim.pathfinder.nav_mesh_settings
+    modified_settings.agent_radius += 0.05
+    assert sim.pathfinder.nav_mesh_settings != modified_settings
+    assert sim.recompute_navmesh(
+        pf, modified_settings
+    ), "failed to recompute navmesh"
+    # Init the shortest path
+    path = habitat_sim.ShortestPath()
+    # Set the locations
+    path.requested_start = snapped_start_pos
+    path.requested_end = snapped_goal_pos
+    # Find the path
+    pf.find_path(path)
+    curr_path_points = path.points
+    if len(curr_path_points) <= 2:
+        return 1.0
```

Comment:
Why do you consider 1.0 to be a failure value? Should we explicitly return a bool to indicate failure? Or perhaps None? 1.0 does not seem like a failure value.

Reply:
I refactored this to be a boolean instead.